added editor.value() method as both setter and getter

jywarren opened the pull request:
Followup on using `editor.value()` to get/set editor content for all three modes in this issue: https://github.com/bevacqua/woofmark/issues/15

Manually tested; i didn't see a test suite but would be happy to add a test if there is one. Thanks!

jywarren committed:
added editor.value() method as both setter and getter

update to parse user input for value() setting

bevacqua commented:
Awesome stuff. I would rather parse the user input. As long as `arguments.length === 1`, use `String(text)` to set the value to their input. In case of HTML mode, convert the provided input into HTML by way of `parseHTML`. Only accepts Markdown (for a simpler API).

Needs docs update

jywarren commented:
OK, updated, I think; can you confirm that I've used `o.parseHTML(text)` correctly? This would still allow setting the raw HTML in `html` mode, but only because we haven't disallowed it. I updated the docs too. 

Thanks!

bevacqua commented:
See b3f25451c0c940a6477417fba13241795f089c43
